src/algebra/monoid_algebra.lean

```diff
@@ -75,6 +74,41 @@ instance : has_mul (monoid_algebra k G) :=
 lemma mul_def {f g : monoid_algebra k G} :
   f * g = (f.sum $ λa₁ b₁, g.sum $ λa₂ b₂, single (a₁ * a₂) (b₁ * b₂)) :=
 rfl
+
+instance : distrib (monoid_algebra k G) :=
+{ mul           := (*),
+  add           := (+),
+  left_distrib  := assume f g h, by simp only [mul_def, sum_add_index, mul_add, mul_zero,
+    single_zero, single_add, eq_self_iff_true, forall_true_iff, forall_3_true_iff, sum_add],
+  right_distrib := assume f g h, by simp only [mul_def, sum_add_index, add_mul, mul_zero, zero_mul,
```

Comment:
I think you can drop `mul_zero` here.

Reply:
I haven't touched any of the proofs in this PR, I just moved them around.